### PR TITLE
Old, modified way of deploying repository

### PR DESCRIPTION
**Pull request by rzadp**
- This PR reverts the changes done in #195 , with modification to make it work for forked PRs.
- Closes https://github.com/poanetwork/tokenbridge/issues/200
- Closes https://github.com/poanetwork/tokenbridge/issues/202
- Note that this is a forked PR

### CODE REVIEW

File: deployment-e2e/molecule/repo/converge.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@
   hosts: all
   become: true
   tasks:
+    - import_role:
+        name: ../../../deployment/roles/common
+        tasks_from: dependencies
```
Comment (rzadp) [inline deployment-e2e/molecule/repo/converge.yml]: Need to add dependencies in this test scenario because otherwise there is no `git` installed on host machine

File: deployment/roles/common/tasks/repo.yml [hunk 2]
```diff
@@ -1,34 +1,21 @@
...
-    - monorepo.tar.gz
-    - contracts.tar.gz
-
-- name: Untar the archives
+- name: Checkout proper branch
   shell: |
-    tar zxfv monorepo.tar.gz && rm monorepo.tar.gz
-    mkdir -p contracts && tar zxfv contracts.tar.gz -C ./contracts && rm contracts.tar.gz
+    # If this is a forked PR, the branch needs to be fetched in a special way
+    [ ! "CIRCLE_PR_NUMBER" = "" ] && git fetch --force origin -- "{{ bridge_repo_branch }}/head:remotes/origin/{{ bridge_repo_branch }}"
```
Comment (rzadp) [inline deployment/roles/common/tasks/repo.yml]: `CIRCLE_PR_NUMBER` is only available on forked PR. So if it is available, we fetch the forked branch in a special way.

File: deployment-e2e/molecule.sh [hunk 3]
```diff
@@ -2,8 +2,10 @@
 cd $(dirname $0)
 set -e # exit when any command fails
 
+CODEBASE_BRANCH=${CIRCLE_BRANCH-$(git symbolic-ref --short HEAD)}
```
Comment (akolotov) [inline deployment-e2e/molecule.sh]: will it still be possible to run `e2e` deployment tests locally?

File: deployment/roles/common/tasks/repo.yml (hunk 2 above)
Comment (akolotov) [inline deployment/roles/common/tasks/repo.yml]: does it make sense to limit the number of downloaded artifacts by `--depth` parameter?

File: deployment-e2e/molecule.sh (hunk 3 above)
Comment (rzadp) [reply deployment-e2e/molecule.sh]: Good question.
- In a normal repository - yes
- In a forked repository - I will work on it now and make sure that it is possible

File: deployment/roles/common/tasks/repo.yml (hunk 2 above)
Comment (rzadp) [reply deployment/roles/common/tasks/repo.yml]: Good idea. Done in https://github.com/poanetwork/tokenbridge/pull/205/commits/a4caf18dcc33c33f14f92fc762e3488588159091

**Comment by rzadp** [comment]
@akolotov I have trouble making it work on forked PRs. Turns out, the tests were passing before for the wrong reason. I found that I had two branches named the same way on both forked and original repository, and it was actually checking out the wrong one!

My current opinion is that approach to `git clone` is arguably more complex than the approach we have currently with tar archives.

When running `deployment-e2e` tests, and also when manually testing deployment on DigitalOcean/AWS, the approach to `git clone` needs to take into account 4 cases:

1. Locally on original repository
    - The local branch needs to be pushed to remote (in order to be `git cloned` later)
    - Need to make sure that in fact given branch is checked out and tested, and not the default `master`
2. CI testing PR from original repository
    - Need to make sure that in fact given branch is checked out and tested
3. Locally on forked repository
   - The local branch needs to be pushed to remote
    - Need to make sure that in fact given branch is checked out and tested
    - Need to make sure that in fact the forked repository is tested, and not the default `poanetwork/tokenbridge`
4. CI testing PR from forked repository
    - Need to make sure that in fact given branch from forked repository is checked out and tested
    - This branch must be fetched and checked-out in a special way

If something gets wrong, it is easy to overlook, because the codebase from original repository's `master` has high chance to just work. Just like I overlooked it with two branches.


I think that our current approach is better and simpler. The tasks to tar/untar are complex, but the principles are simple:

1. Locally, the code you see is deployed / tested. No matter if you have original repository, forked repository, or if you downloaded repository `.zip`
2. When testing original/forked PR, CI checks out proper code. This code gets tested.

In conclusion, I argue that it would be better to leave our current approach with copying repository as it is.
@akolotov What do you think?

**Comment by akolotov** [comment]
> I think that our current approach is better and simpler. The tasks to tar/untar are complex, but the principles are simple:
> 
> 1. Locally, the code you see is deployed / tested. No matter if you have original repository, forked repository, or if you downloaded repository `.zip`
> 2. When testing original/forked PR, CI checks out proper code. This code gets tested.
> 
> In conclusion, I argue that it would be better to leave our current approach with copying repository as it is.
> @akolotov What do you think?

Thanks @rzadp for detailed explanation. Please add this comment to the issue that is opened to investigate possibility to switch back to `git clone`.

**Comment by akolotov** [comment]
If nothing is going to be changed as part of this PR, please close it.
